tests/frontier/blockchain_st_test_helpers.py

```diff
@@ -76,6 +76,13 @@ def run_blockchain_st_test(
 
 def add_blocks_to_chain(chain: BlockChain, test_data: Dict[str, Any]) -> None:
     for idx, block in enumerate(test_data["blocks"]):
+        if idx == 0:
+            assert test_data["genesis_header_hash"] == block.header.parent_hash
```

Comment:
These assert statements basically check that the test data is correct. This doesn't ensure that the `spec.py` actually validates the parent hash.
What we should be doing is writing a unit test that provides an invalid parent hash and checks that test reverts with a validation error.

Reply:
@voith Thank you for the input. I have a few questions here.

I see that the invalid blocks test cases are already present under tests/fixtures/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcInvalidHeaderTest/

I don't think any of these test cases are implemented yet (at least I couldn't find the implementations). Is that right? If I've missed it, could you please point me to the implementation?

If they have not been implemented yet, please suggest the file where adding these tests would be most appropriate. (test_state_transition.py? or maybe a new file?)

Reply:
> I see that the invalid blocks test cases are already present under tests/fixtures/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcInvalidHeaderTest/


Some of those tests were added sometime back and were accidentally removed by me. Please refer to #415 for more context. You can add them back by referring to the commit I've posted in that issue. 
However, only [bcUncleTest](https://github.com/ethereum/tests/tree/82f6a5a7362e703a64dce303dd0c30f264b8195f/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcUncleTest)  and [bcUncleHeaderValidity](https://github.com/ethereum/tests/tree/82f6a5a7362e703a64dce303dd0c30f264b8195f/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcUncleHeaderValidity) were tested earlier. You probably need [bcInvalidHeaderTest](https://github.com/ethereum/tests/tree/82f6a5a7362e703a64dce303dd0c30f264b8195f/LegacyTests/Constantinople/BlockchainTests/InvalidBlocks/bcInvalidHeaderTest) in order to close #387

Reply:
@voith Accidentally happened to remove the repo from which I had raised the PR. Have opened a new PR with updates to include the relevant unit tests. Please review

https://github.com/ethereum/execution-specs/pull/416